kratos/geometries/quadrilateral_2d_8.h

```diff
@@ -1534,41 +1534,41 @@ template<class TPointType> class Quadrilateral2D8
         for ( unsigned int pnt = 0; pnt < integration_points_number; pnt++ )
         {
             // Primary nodes
-            row( shape_function_values, pnt )( 0 ) =
+            row( shape_function_values, pnt )[0] =
                 -(( 1.0 - integration_points[pnt].X() )
                   * ( 1.0 - integration_points[pnt].Y() )
                   * ( 1.0 + integration_points[pnt].X()
                       + integration_points[pnt].Y() ) ) / 4.0;
-            row( shape_function_values, pnt )( 1 ) =
+            row( shape_function_values, pnt )[1] =
                 -(( 1.0 + integration_points[pnt].X() )
                   * ( 1.0 - integration_points[pnt].Y() ) * ( 1.0
                           - integration_points[pnt].X()
                           + integration_points[pnt].Y() ) ) / 4.0;
-            row( shape_function_values, pnt )( 2 ) =
+            row( shape_function_values, pnt )[2] =
                 -(( 1.0 + integration_points[pnt].X() )
                   * ( 1.0 + integration_points[pnt].Y() ) * ( 1.0
                           - integration_points[pnt].X()
                           - integration_points[pnt].Y() ) ) / 4.0;
-            row( shape_function_values, pnt )( 3 ) =
+            row( shape_function_values, pnt )[3] =
                 -(( 1.0 - integration_points[pnt].X() ) * ( 1.0
                         + integration_points[pnt].Y() ) * ( 1.0 ) * ( 1.0
                         + integration_points[pnt].X()
                         - integration_points[pnt].Y() ) ) / 4.0;
             
             // Secondary nodes
-            row( shape_function_values, pnt )( 4 ) =
```

Comment:
@pooyan-dadvand why did u decide not to support access operator by (i) ? asking i guess this will be the most painful change to do

Reply:
With current implementation of AMatrix I should add it to all the expressions and there weren't so many places in the code (core at least) which using it. It also gives me some problem when to define it or not. 

Also I don't like it because is not coherent with normal arrays but of course this is my personal opinion.

Reply:
i am afraid this is going to be very painful in the applications since the notation Vec(i) is used quite a lot.